compiler/src/dotty/tools/dotc/typer/Applications.scala

```diff
@@ -1148,22 +1148,11 @@ trait Applications extends Compatibility {
     record("typedTypeApply")
     typedExpr(tree.fun, PolyProto(typedArgs, pt)) match {
       case fun: TypeApply if !ctx.isAfterTyper =>
-        if !Feature.clauseInterleavingEnabled then
-          errorTree(tree, em"illegal repeated type application")
-        else
-          // We want to be able to call `def f[A](using b: B[A])[C <: b.type]` as `f[MyA][MyC]`
-          // However this is fundamentally incompatible with the way named type arguments currently work
-
-          // Therefore when named type arguments are reworked, this implementation restriction will have to be removed
-          // to be up to date with the spec as described in SIP-47 - Clause Interleaving
-          
-          // The explicit call to namedTypeArgsEnabled below ensures this comment will not be forgotten when the time comes
-          Feature.namedTypeArgsEnabled
-          val function = fun.fun
-          val args = (tree.args ++ fun.args).map(_.show).mkString(", ")
-          errorTree(tree, em"""implementation restriction: repeated type application is not yet supported. 
-                             |If you meant for a using clause to be implied between the two type clauses, please write it out explicitly.
-                             |You might also have meant something like ${function}[${args}]""")
+        val function = fun.fun
+        val args = (tree.args ++ fun.args).map(_.show).mkString(", ")
+        errorTree(tree, em"""illegal repeated type application
```

Comment:
I'm not digging too deep here, but its possible to have repeated type application in general, e.g. curried type lambdas.
```scala
type Foo = [A] =>> [B] =>> Map[A, B]
val x: Foo[String][Int] = Map("abc" -> 123)
```

Reply:
yes you're right :) (`AppliedTypeTree`)